http3/roundtrip.go

```diff
@@ -165,7 +165,10 @@ func (r *RoundTripper) RoundTripOpt(req *http.Request, opt RoundTripOpt) (*http.
 	defer cl.useCount.Add(-1)
 	rsp, err := cl.rt.RoundTripOpt(req, opt)
 	if err != nil {
-		r.removeClient(hostname)
+		if !errors.Is(err, context.Canceled) {
```

Comment:
Add a comment here why you're not removing the connection for context cancelations?